### PR TITLE
Add tests for `NoPhpOpenTagInCodeBlockPhpDirective` rule

### LINKED ISSUE #1299: Write tests for `NoPhpOpenTagInCodeBlockPhpDirective` rule

### PR DESCRIPTION
**Pull request by MrYamous**
Fixes #1299

### CODE REVIEW

**Comment by MrYamous** [comment]
I don't understand how to deal with last value from `UnitTestCase::phpCodeBlocks`, could you help me ?

File: tests/Rule/NoPhpOpenTagInCodeBlockPhpDirectiveTest.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        static::assertEquals(
+            $expected,
+            (new NoPhpOpenTagInCodeBlockPhpDirective())->check($sample->lines(), $sample->lineNumber(), 'filename')
+        );
+    }
+
+    public function checkProvider(): \Generator
+    {
+        foreach (self::phpCodeBlocks() as $codeBlock) {
+            var_dump($codeBlock);
```
Comment (OskarStark) [inline tests/Rule/NoPhpOpenTagInCodeBlockPhpDirectiveTest.php:40]: 🤓

**Comment by OskarStark** [comment]
I will have a look tomorrow 🙂

File: tests/Rule/NoPhpOpenTagInCodeBlockPhpDirectiveTest.php (hunk 1 above)
Comment (MrYamous) [reply tests/Rule/NoPhpOpenTagInCodeBlockPhpDirectiveTest.php:40]: Ooops

File: src/Rule/NoPhpOpenTagInCodeBlockPhpDirective.php [hunk 2]
```diff
@@ -35,7 +35,7 @@ public function check(Lines $lines, int $number, string $filename): ViolationInt
         $lines->seek($number);
         $line = $lines->current();
 
-        if (!RstParser::codeBlockDirectiveIsTypeOf($line, RstParser::CODE_BLOCK_PHP, true)
+        if (!RstParser::codeBlockDirectiveIsTypeOf($line, RstParser::CODE_BLOCK_PHP)
```
Comment (OskarStark) [inline src/Rule/NoPhpOpenTagInCodeBlockPhpDirective.php:38]: > I don't understand how to deal with last value from UnitTestCase::phpCodeBlocks, could you help me ?

Congrats 🎉 you found a bug!

**Pull request merged**